### PR TITLE
i#3544 RV64: Enable sample.bbbuf test

### PR DESCRIPTION
**Pull request by ksco**
This patch enables sample.bbbuf test with the following fixes:

- x0 on RISC-V is like XZR on AArch64, so do not consider it as a general register
- tp should be avoided in `drreg_reserve_reg_internal()`
- RISC-V does not have control flags, so a dummy hello test is used for `control_flags` instead

Also removed `linux.sigaction_nosignals` and `pthreads.pthreads` from `RUNS_ON_QEMU` since they are not working on real hardware.

#3544

### CODE REVIEW

File: suite/tests/CMakeLists.txt [hunk 1]
```diff
@@ -5910,10 +5911,9 @@ if (RISCV64)
     code_api|linux.infinite
     code_api|linux.longjmp
     code_api|linux.prctl
-    code_api|linux.sigaction_nosignals
```
Comment (abhinav92003) [inline suite/tests/CMakeLists.txt:5913]: They're not working on real hardware but they're working on QEMU? How? Would be good to keep the QEMU run if they're indeed working fine there.

**Review by abhinav92003 — APPROVED** [review]

File: suite/tests/CMakeLists.txt (hunk 1 above)
Comment (ksco) [reply suite/tests/CMakeLists.txt:5913]: > if they're indeed working fine there

It's strange, so I'm not sure if they're working as expected, but will add them back once figure out why.

**Comment by ksco** [comment]
@abhinav92003 Can you merge this? Thanks.

**Comment by abhinav92003** [comment]
x86-32 failures are #6417.

**Pull request merged**